library/core/qsl-base/src/main/java/org/quiltmc/qsl/api/event/ParameterInvokingEvent.java

```diff
@@ -0,0 +1,79 @@
+package org.quiltmc.qsl.api.event;
+
+import java.lang.annotation.Documented;
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+
+/**
+ * A marker interface that indicates an event stored in a field or returned by a method may invoke the event callback on
+ * one or more context objects passed in some method parameters.
+ *
+ * <p>The primary use of this annotation is for static analysis and documentation. Where a field or method return value
+ * is annotated as a parameter invoking, you are encouraged to note in the documentation which context parameters may be
+ * invoked.
+ *
+ * <p>Event implementations may have an undefined order in which the parameter may be invoked, see the documentation of
```

Comment:
Is this what you mean?
```suggestion
 * <p>Event implementations may have an undefined order in which the parameters may be invoked, see the documentation of
```